### ISSUE TITLE
Various (memory leak?) errors when reasoning over different KBs

### ISSUE THREAD

**Issue by mrnolte** [knowsys/rulewerk#177]
Hey there,
Consider the 4 kbs contained in [this zip file](https://github.com/knowsys/rulewerk/files/4689931/error.zip). After reasoning over the first three, on the forth I get various errors that make the jvm crash (changing every time I execute the code), including the following:
"double free or corruption (fasstop)"
"pure virtual function call"
"corrupted size vs prev_size"
Sometimes, I even get the error message and core dump contained in [this log file](https://github.com/knowsys/rulewerk/files/4689969/hs_err_pid2548.log).

This is my code:
```java
for (int i = 0; i < 4; i++) {
	final KnowledgeBase kb = RuleParser
			.parse(new FileInputStream(Paths.get("error_" + i + ".txt").toFile()));
	try (final Reasoner reasoner = new VLogReasoner(kb)) {
		if (!reasoner.reason()) {
			throw new RuntimeException();
		}
	}
}
```

Interestingly, these errors do not occur when reasoning over the fourth KB only.
Note that I use Ubuntu-WSL, but usually it works just fine (detailed system infos are included in the dump file).

Cheers,
Robin

**Comment by CerielJacobs** [comment]
Hi,
this sounds like a VLog bug, so I tried this (after removing an _, since the input filenames don’t have this) with the current master, but I don’t get past line 1 of error0.txt:

org.semanticweb.rulewerk.parser.javacc.TokenMgrError: Lexical error at line 1, column 101.  Encountered: "R" (82), after : “_"

Maybe someone more knowledgeable on Rulewerk can help me to get going with this?

Best,   Ceriel

**Comment by mmarx** [comment]
I'll take a look this afternoon.

**Comment by mrnolte** [comment]
Ah @CerielJacobs, I am sorry; I did upload the wrong ontologies. Rulewerk (without a warning) lets you create KBs, that when exported using `KnowledgeBase.writeKnowledgeBase(String)` can't be parsed again because of unsuported chars. You can find the corrected ones [here](https://github.com/knowsys/rulewerk/files/4694770/error.zip) (my renaming confused myself, it seems).

**Comment by mkroetzsch** [comment]
@mrnolte Independent of this bug, we should look into the syntactic roundtripping issue that you mentioned. Our goal is to fully preserve KBs that use reasonable characters and to make sure that cases where this cannot be achieved are at least parseable. Feel free to submit an issue with the specific encoding issues you encountered.

**Comment by mmarx** [comment]
It's probably related to #155, and similar problems for other named constructs such as Variables and Predicate names.

**Comment by CerielJacobs** [comment]
On my Macbook it runs to completion. On my Ubuntu Linux VM, it indeed crashes. This may be a hard one to find ...

**Comment by mmarx** [comment]
I can reproduce the crash as well.

**Comment by mrnolte** [comment]
Hey there,
Unfortunately I did not check the license first before uploading the KBs. They were automatically generated from a public ontology, but since the ontology did not include a license, I had to have the zip-files removed, as I am not sure if I may distribute them here in an edited form. I contacted the ontology publisher and will reupload them if i have the license to do so. Otherwise I am going to open a repo that contains code to generate the KBs from scrap within the next few days. I'm sorry for the inconvenience!
